packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -57,10 +58,20 @@ export type TabsConfig = { type: "tabs"; tabs: { name: string; content: Generato
 
 type DataGridSettings = Pick<GridColDef, "headerName" | "width" | "minWidth" | "maxWidth" | "flex">;
 
+export type StatusMapping = Record<
+    string,
+    {
+        color: SvgIconTypeMap["props"]["color"];
+        primaryText: string;
+        secondaryText?: string;
+    }
+>;
+
 export type GridColumnConfig<T> = (
     | { type: "text" }
     | { type: "number" }
     | { type: "boolean" }
+    | { type: "status"; statusMapping: StatusMapping }
```

Comment:
I'm not sure about the naming of this new type, isn't this the same as a staticSelect? (which also has enums)

Reply:
Thats a good point 👍🏼 

What do you think of extending `staticSelect` to allow overriding the `valueOptions`?

Then we could also support `number` and `boolean` values (the `inStock` example for instance requires booleans). 
Also the label could include any icon, instead of just a string, if necessary, by rendering `GridCellContent`. 

Then the config could be like this for example:

```ts
// Config types in generator
type CometAdminIconKey = string; // Can/should we limit this to exports of "@comet/admin-icons"?

type LabelIconObject = {
    name: CometAdminIconKey;
    color?: string;
};

type LabelObject = {
    icon?: CometAdminIconKey | LabelIconObject | ImportReference;
    primaryText?: string;
    secondaryText?: string;
};

type StaticSelectValueOption = {
    value: string | number | boolean;
    label: string | LabelObject;
};

// Config example
{
    type: "staticSelect",
    name: "inStock",
    headerName: "In Stock",
    valueOptions: [
        {
            value: true,
            label: {
                icon: { name: "StateFilled", color: "success" },
                primaryText: "In Stock",
            },
        },
        {
            value: false,
            label: {
                icon: { name: "StateFilled", color: "error" },
                primaryText: "Out of Stock",
            },
        },
    ],
},
```

Then this would be similar to the already generated, non-overridable `valueOptions` from e.g. the "type" `staticSelect` in the demo `ProductsGrid`:

https://github.com/vivid-planet/comet/blob/89a3e367b686782a2eba6261cb3de79431c8a04c/demo/admin/src/products/future/generated/ProductsGrid.tsx#L119-L126

Reply:
staticSelect already supports `values?: string[]`, maybe we could facilitate that?

But I think it's a good idea to allow overriding the label and add an icon.

Reply:
> staticSelect already supports `values?: string[]`

Well... 😄 

https://github.com/vivid-planet/comet/blob/47e04d9c1cc484d6c38946803c8199b0d7dfbaeb/packages/admin/cms-admin/src/generator/future/generateGrid.ts#L250-L252

`values?: string[]` doesn't really make sense to me, as there is no way to match the passed in string with the values defined by the api...

> But I think it's a good idea to allow overriding the label and add an icon.

Should we use the `columns.values` config for that somehow or should we add a `valueOptions` object instead, like the generated one?

Reply:
see https://github.com/vivid-planet/comet/pull/2265

Reply:
@nsams should I use your PR as a base for mine, and extend the `label` similar to my suggestion above?

Reply:
yes; i have to finish it then asap...